### ISSUE TITLE
Python Decorator written below @brief tag and above the function leads to documentation error

### ISSUE THREAD

**Issue by supriyaKrishnamurthy** [doxygen/doxygen#8098]
**In Python files if a decorator is used for function or class after @brief, then it doesnt consider the brief tag and end up throwing error.But making it write above the @brief tag looks odd.Can you please suggest the solution to this**

![doxygen_decorator_issue](https://user-images.githubusercontent.com/7384828/95959457-1d678d00-0e20-11eb-9fc8-7d8029bbf7ab.png)

**Expected behavior**
It should not throw error/warning

@albert-github  @doxygen  please help me resolve this

**Comment by supriyaKrishnamurthy** [comment]
![doxygen_decorator_issue](https://user-images.githubusercontent.com/7384828/95959285-e002ff80-0e1f-11eb-8ebf-e4d03f974460.png)

**Comment by albert-github** [comment]
Decorators are not handled in doxygen and to the best of my knowledge there isn't a list of decorators in python. In the code I see that the decorator `@staticmethod` is handled though.

When adding code or text to an issue don't add it as an image but as text, now it is very hard to use it, so
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.  
- Please also specify the full doxygen version used (`doxygen -v`).

**Comment by supriyaKrishnamurthy** [comment]
Ya Sure. The version I am using is 1.8.20 and please find the attachment of sample code and doxyfile used.
[doxygen.zip](https://github.com/doxygen/doxygen/files/5380420/doxygen.zip)

**Comment by supriyaKrishnamurthy** [comment]
Hi @albert-github @doxygen any update on this. Or is there a way to ignore these decorators while checking for documentaion in python.

**Comment by albert-github** [comment]
A little bit impatient ....

Best you can do at the moment is to write the enhancement yourself for handling all types of decorators in doxygen for python code and create a pull request for it.
A workaround is to write a filter (see `FILTER_PATTERNS`) that filters out the decorators.

**Comment by supriyaKrishnamurthy** [comment]
Can you give one example of how to use FILTER_PATTERNS to ignore decorators

**Comment by supriyaKrishnamurthy** [comment]
I tried using @cond and @endcond to make doxygen ignore the processing of part of the code that contains decorator but it is not working.

**Comment by albert-github** [comment]
When I use:
```
## \file

##
# @brief Because of the below decorator it says function not documented even if its documented
# @return None
# @cond TEST
@common.configure_test(repeat=True, selective=["abc", "xyz"])
## @endcond
def foo(self):
    self.somefunction()
```
it does work, but one should not add `TEST` to `ENABLED_SECTIONS` as at that moment the `@cond` results in the fact that the part is included (analogous to e.g. the `#if` in C).

Note that in the `doxyfile` you provided no output format has been selected by means of the `GENERATE_..`

**Comment by supriyaKrishnamurthy** [comment]
yea for some testing purpose I had disable GENERATE_HTML. I enabled it and I left ENABLED_SECTIONS blank and tried. It Worked!!!! Cant thank you enough @albert-github .This is a great help for me. Sorry for being impatient, and thans a lot for the help

**Comment by doxygen** [comment]
Added minimal support for decorators. They should now be parsed correctly, but do not appear in the documentation. Please verify if it solves the problem without the need for workarounds.

**Comment by supriyaKrishnamurthy** [comment]
Thank you addressing my issue in your tool, can you share the steps to follow.Do I need to need to update the version ?

**Comment by albert-github** [comment]
You can either wait for a new release of get the latest master version and build it yourself (there are no intermediate build available).

**Comment by supriyaKrishnamurthy** [comment]
Okay Thanku

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
